appserver/payara-appserver-modules/microprofile/healthcheck/src/main/java/fish/payara/microprofile/healthcheck/HealthCheckService.java

```diff
@@ -191,6 +202,12 @@ public void performHealthChecks(HttpServletResponse response) throws IOException
             }
         }
 
+        // No applications (yet), server is not ready.
+        if (applicationsLoaded.isEmpty()) {
+            // Application is not yet deployed
+            healthCheckResponses.add(HealthCheckResponse.builder().down().build());
```

Comment:
Is this valid? It should at least have a name.

Reply:
I can't find anything in the spec that the name is required. But I added a name.